src/xregexp.js

```diff
@@ -791,21 +793,28 @@ XRegExp.escape = function(str) {
 XRegExp.exec = function(str, regex, pos, sticky) {
     var cacheKey = 'g',
         addY = false,
+        fakeY = false,
         match,
         r2;
 
     addY = hasNativeY && !!(sticky || (regex.sticky && sticky !== false));
     if (addY) {
         cacheKey += 'y';
+    } else if (sticky) {
+        cacheKey += 'fakeY'
+        fakeY = true
```

Comment:
BTW, I'm not entirely satisfied by the details of this patch. Having a FakeY-specific pathway in `copyRegex` is suboptimal... Maybe we could instead honour `options.source`. `XRegExp.exec` could then inject `source: regex.source + '|()'`...